_posts/2023-09-25-Debezium-JMX-signaling-and-notifications.adoc

```diff
@@ -0,0 +1,1558 @@
+    "database.port": "5432",
+    "database.user": "postgres",
+    "database.password": "postgres",
+    "database.server.id": "184054",
+    "database.dbname": "postgres",
+    "topic.prefix": "dbserver1",
+    "snapshot.mode": "NEVER",
+    "schema.history.internal.kafka.bootstrap.servers": "kafka:9092",
+    "schema.history.internal.kafka.topic": "schema-changes.inventory",
+    "signal.enabled.channels": "source,jmx", <1>
+    "signal.data.collection": "inventory.debezium_signal", <2>
+    "notification.enabled.channels": "jmx"
+  }
+}
+----
+<1> This configuration enables **source** and **jmx** channel. Even if we will use JMX to send the signal, since we want to execute an incremental snapshot, the **source** signal is still required because Debezium needs to use signal table to watermark the db log for events decoupling.
+<2> Set the table used for the signaling
+
+[NOTE]
+For now, don't worry about the **notification.enabled.channels** property. We will delve into it later
+
+After saving this configuration into a file named **postgres-jmx.json**, we can register it.
+
+To register the connector we can use `curl` to call the Kafka Connect API
+
+[source, bash]
+----
+curl -i -X POST -H "Accept:application/json" -H "Content-Type:application/json" localhost:8083/connectors/ -d '{"name":"inventory-connector","config":{"connector.class":"io.debezium.connector.postgresql.PostgresConnector","tasks.max":"1","database.hostname":"postgres","database.port":"5432","database.user":"postgres","database.password":"postgres","database.server.id":"184054","database.dbname":"postgres","topic.prefix":"dbserver1","snapshot.mode":"NEVER","schema.history.internal.kafka.bootstrap.servers":"kafka:9092","schema.history.internal.kafka.topic":"schema-changes.inventory","signal.enabled.channels":"source,jmx","signal.data.collection":"inventory.debezium_signal","notification.enabled.channels":"log,sink,jmx","notification.sink.topic.name":"io.debezium.notification"}}'
+----
+
+or I would suggest to use **https://github.com/kcctl/kcctl[kcctl]** tool to interact with Kafka Connect. It is a modern and intuitive command line client for Kafka Connect.
+
+First of all we need to create a configuration context to connect with Kafka Connect
+
+[source, bash]
+----
+kcctl config set-context local --cluster http://localhost:8083
+----
+
+then we can register the connector running the following command
+
+[source, bash]
+----
+kcctl apply -f postgres-jmx.json
+----
+
+we can now get the logs of connect container
+
+[source, bash]
+----
+docker logs connect
+----
+
+and check that the connector is started streaming events
+
+[source, text]
+----
+INFO   Postgres|dbserver1|streaming  Starting streaming   [io.debezium.pipeline.ChangeEventSourceCoordinator]
+----
+
+=== Prepare database for incremental snapshot
+Since incremental snapshot still requires to have the `signal.data.collection` we need to create the signal table on our postgres database.
```

Comment:
```suggestion
Since incremental snapshots require the `signal.data.collection` to be defined, we need to create the signal table on our postgres database.
```